packages/itwin/grouping-mapping-widget/src/widget/components/GroupAction.tsx

```diff
@@ -133,43 +164,14 @@ const GroupAction = ({
     );
   }, [iModelConnection]);
 
-  const isWrappedInQuotes = (text: string) => {
-    return text.startsWith(`"`) && text.endsWith(`"`);
-  };
-
-  const needsAndOperator = (token: string, index: number, searchQuery: string[]) => isWrappedInQuotes(token) || (index === 1 && isWrappedInQuotes(searchQuery[0]));
-  // Temporary until ECViews become available for use.
-  const generateSearchQuery = (searchQuery: string[]) => {
-    if (searchQuery.length === 0) {
-      setQuery("");
-      return;
-    }
-
-    let generatedSearchQuery =
-      `SELECT be.ECInstanceId, be.ECClassId FROM bis.geometricelement3d be `;
-    generatedSearchQuery += `LEFT JOIN bis.SpatialCategory cat ON be.Category.Id = cat.ECInstanceID LEFT JOIN ecdbmeta.ECClassDef ecc ON be.ECClassId = ecc.ECInstanceId `;
-    generatedSearchQuery += `LEFT JOIN bis.PhysicalType pt ON be.TypeDefinition.Id = pt.ECInstanceID`;
-    generatedSearchQuery += ` WHERE `;
-    generatedSearchQuery += `((${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} be.codevalue LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")}) OR (${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} be.userlabel LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")})) OR ((${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} cat.codevalue LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")}) OR (${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} cat.userlabel LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")})) OR (${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} ecc.name LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")})`;
-    // Physical Types
-    generatedSearchQuery += ` OR ((${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} pt.codevalue LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")}) OR (${searchQuery.map((token, index) =>
-      `${index === 0 ? "" : needsAndOperator(token, index, searchQuery) ? "AND" : "OR"} pt.userlabel LIKE '%${isWrappedInQuotes(token) ? token.slice(1, -1) : token}%'`
-    ).join(" ")})) `;
+  const updateQuery = useCallback(
```

Comment:
This callback is redundant. You can pass around the `setQuery` function. It is as stable as the one you create here.

Reply:
outdated

Reply:
No, it's not outdated.